src/pages/preferences/preferencesTabs/publicKeysTab/EditModal.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useEffect } from "react";
+import { Modal } from "components/Modal";
+import { useMutation } from "@apollo/react-hooks";
+import styled from "@emotion/styled";
+import Button, { Variant } from "@leafygreen-ui/button";
+import {
+  GetMyPublicKeysQuery,
+  UpdatePublicKeyMutation,
+  UpdatePublicKeyMutationVariables,
+  GetMyPublicKeysQueryVariables,
+  CreatePublicKeyMutation,
+  CreatePublicKeyMutationVariables,
+} from "gql/generated/types";
+import { GET_MY_PUBLIC_KEYS } from "gql/queries";
+import { UPDATE_PUBLIC_KEY } from "gql/mutations/update-public-key";
+import { useBannerDispatchContext } from "context/banners";
+import { Input } from "antd";
+import { CREATE_PUBLIC_KEY } from "gql/mutations/create-public-key";
+import { InputLabel, ErrorMessage } from "components/styles";
+import { v4 as uuid } from "uuid";
+
+const { TextArea } = Input;
+
+export interface EditModalPropsState {
+  replaceKeyName?: string;
+  initialKeyName?: string;
+  initialKeyValue?: string;
+  visible: boolean;
+  existingKeys?: {
+    name: string;
+    key?: string;
+  }[];
+}
+
+interface EditModalProps extends EditModalPropsState {
+  onCancel: () => void;
+}
+
+export const EditModal: React.FC<EditModalProps> = ({
+  replaceKeyName,
+  initialKeyName,
+  initialKeyValue,
+  existingKeys,
+  visible,
+  onCancel,
+}) => {
+  const dispatchBanner = useBannerDispatchContext();
+  const [errors, setErrors] = useState<string[]>([]);
+  const [updatePublicKey] = useMutation<
+    UpdatePublicKeyMutation,
+    UpdatePublicKeyMutationVariables
+  >(UPDATE_PUBLIC_KEY, {
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error editing the public key: ${error.message}`
+      );
+    },
+    update(cache, { data }) {
+      cache.writeQuery<GetMyPublicKeysQuery, GetMyPublicKeysQueryVariables>({
+        query: GET_MY_PUBLIC_KEYS,
+        data: { myPublicKeys: [...data.updatePublicKey] },
+      });
+    },
+  });
+  const [createPublicKey] = useMutation<
+    CreatePublicKeyMutation,
+    CreatePublicKeyMutationVariables
+  >(CREATE_PUBLIC_KEY, {
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error creating the public key: ${error.message}`
+      );
+    },
+    update(cache, { data }) {
+      cache.writeQuery<GetMyPublicKeysQuery, GetMyPublicKeysQueryVariables>({
+        query: GET_MY_PUBLIC_KEYS,
+        data: { myPublicKeys: [...data.createPublicKey] },
+      });
```

Comment:
I think from a readability perspective this is confusing. Lets just call refetchQueries when we perform any mutation like we do elsewhere in the code base. Manually updating the apollo cache is error prone and in this case this is very cheap query so i dont see any advantage to updating the cache manually.

Reply:
i agree that refetchQueries will do fine here but i don't find this confusing. what about writing to the cache is error prone? all of our queries being statically typed makes it less error prone.

Reply:
this requires a little more code than using using `refetchQueries` but is documented well in the Apollo docs so I'm too concerned about readability. I think the tedious/error prone part is that there are 3 different gql endpoints that I use to get a list of public keys and they all have to be maintained. If I used `refetchQueries`, we only rely on a single query to get all of a users public keys.

Reply:
i do think that refetchQueries is better for this query, but i don't want to rule out making writes to the cache. the feature itself is not error prone and works very well. Also, in the case where multiple queries rely on the same field - that is exactly what `fragments` are for

Reply:
@khelif96 I'm going to keep the current implementation since it's working properly and keep in mind `refetchQueries` for future mutations where the payload is simple/small